libraries/AP_Camera/AP_Camera_Logging.cpp

```diff
@@ -19,57 +20,95 @@ void AP_Camera_Backend::Write_CameraInfo(enum LogMessages msg, uint64_t timestam
-    if (gps.status() >= AP_GPS::GPS_OK_FIX_3D) {
-        altitude_gps = gps.location().alt;
+        int32_t altitude, altitude_rel, altitude_gps;
+        if (current_loc.relative_alt) {
+            altitude = current_loc.alt+ahrs.get_home().alt;
+            altitude_rel = current_loc.alt;
+        } else {
+            altitude = current_loc.alt;
+            altitude_rel = current_loc.alt - ahrs.get_home().alt;
+        }
+        const AP_GPS &gps = AP::gps();
+        if (gps.status() >= AP_GPS::GPS_OK_FIX_3D) {
+            altitude_gps = gps.location().alt;
+        } else {
+            altitude_gps = 0;
+        }
+
+        const struct log_Camera pkt1{
+            LOG_PACKET_HEADER_INIT(static_cast<uint8_t>(msg)),
+            time_us     : timestamp_us ? timestamp_us : AP_HAL::micros64(),
+            instance    : _instance,
+            image_number: image_index,
+            gps_time    : gps.time_week_ms(),
+            gps_week    : gps.time_week(),
+            latitude    : current_loc.lat,
+            longitude   : current_loc.lng,
+            altitude    : altitude,
+            altitude_rel: altitude_rel,
+            altitude_gps: altitude_gps,
+            roll        : (int16_t)ahrs.roll_sensor,
+            pitch       : (int16_t)ahrs.pitch_sensor,
+            yaw         : (uint16_t)ahrs.yaw_sensor
+
+        };
+
+        AP::logger().WriteCriticalBlock(&pkt1, sizeof(pkt1));
     } else {
-        altitude_gps = 0;
-    }
+        const auto nanf = AP::logger().quiet_nanf();
+        float_t roll = nanf;
+        float_t pitch = nanf;
+        float_t yaw = nanf;
+        float_t target_roll = nanf;
+        float_t target_pitch = nanf;
+        float_t target_yaw = nanf;
+
+        auto *mount = AP_Mount::get_singleton();
+        if (mount != nullptr) {
+            // get mount's current angles and convert yaw to earth frame
+            mount->get_attitude_euler(roll,pitch,yaw);
+            yaw = wrap_PI(yaw + AP::ahrs().yaw);
 
-    const struct log_Camera pkt{
-        LOG_PACKET_HEADER_INIT(static_cast<uint8_t>(msg)),
-        time_us     : timestamp_us ? timestamp_us : AP_HAL::micros64(),
-        instance    : _instance,
-        image_number: image_index,
-        gps_time    : gps.time_week_ms(),
-        gps_week    : gps.time_week(),
-        latitude    : current_loc.lat,
-        longitude   : current_loc.lng,
-        altitude    : altitude,
-        altitude_rel: altitude_rel,
-        altitude_gps: altitude_gps,
-        roll        : (int16_t)ahrs.roll_sensor,
-        pitch       : (int16_t)ahrs.pitch_sensor,
-        yaw         : (uint16_t)ahrs.yaw_sensor
-    };
-    AP::logger().WriteCriticalBlock(&pkt, sizeof(pkt));
+            // get mount's target angles and convert yaw to earth frame
+            mount->get_target_attitude_euler(target_roll,target_pitch,target_yaw);
+            target_yaw = wrap_PI(target_yaw + AP::ahrs().yaw);
+        }
+
+        const struct log_Mount pkt2{
```

Comment:
Instead of putting the logging here in AP_Camera_Logging.cpp it should really be in AP_Mount.